background.js

```diff
@@ -0,0 +1,471 @@
+		tsServerOptionName + '=' + configOptions[tsServerOptionName] + ', ' +
+		tsProjectOptionName + '=' + configOptions[tsProjectOptionName] + ', ' +
+		sapUserOptionName + '=' + configOptions[sapUserOptionName] + ')';
+	}
+
+	const testId = tabToTestCaseMap[tabId];
+
+	if (action === 'update' && (!extendedName || !status)) {
+		throw 'Need test name and status for updating Test Run.';
+	}
+
+	const request = constructTeamscaleRequest(action, status, extendedName, tabId, sapTestKey, testId);
+	request.setRequestHeader('X-Requested-By', teamscaleSession);
+
+	request.onreadystatechange = () => handleTeamscaleResponse(request, action);
+	request.send();
+}
+
+function constructTeamscaleRequest(action, status, extendedName, tabId, sapTestKey, testId) {
+	const request = new XMLHttpRequest();
+
+	let additionalParameter = '';
+	if (action === 'update') {
+		additionalParameter = '&result=' + status + "&extended-name=" + encodeURI(extendedName);
+	}
+
+	const teamscaleUrl = assertStringEndsWith(configOptions[tsServerOptionName], '/');
+
+	const testOutput = 'Follow this link to view test run in Azure DevOps:\n' + tabToAdosTestRunUrlMap[tabId];
+
+	let url;
+	let httpVerb = 'POST';
+	const serviceUrl = teamscaleUrl + 'api/projects/' + configOptions[tsProjectOptionName] + '/sap-test-event/';
+
+	if (action === 'reset') {
+		url = serviceUrl + action + '/' + encodeURIComponent(configOptions[sapUserOptionName]);
+	} else if (action === 'log') {
+		httpVerb = 'GET';
+		url = serviceUrl + action + '/' + encodeURIComponent(sapTestKey);
+	} else {
+		url = serviceUrl + action + '?test-id=' + testId + '&message=' + encodeURIComponent(testOutput) + '&sap-user-name=' + encodeURIComponent(configOptions[sapUserOptionName]) + additionalParameter;
+	}
+
+	request.open(httpVerb, url, true);
+	return request;
+}
+
+function handleTeamscaleResponse(request, action) {
+	if (request.readyState !== 4) {
+		return;
+	}
+
+	const actionString = resolveActionCaption(action);
+	const text = resolveEventText(request);
+	const event = {
+		action: actionString,
+		status: request.status,
+		date: new Date().toLocaleString(),
+		text: text
+	};
+
+	persistAndBroadcastSingleEvent(event);
+}
+
+function resolveActionCaption(action) {
+	switch (action) {
+		case 'start':
+			return '▶️ Start Test';
+		case 'stop':
+			return '⏹️ End Test';
+		case 'update':
+			return '🗞️ Update Metadata';
+		case 'log':
+			return '🗞️ Log';
+		case 'reset':
+			return '🔄 Reset Recording State';
+		default:
+			return 'Test Event';
+	}
+}
+
+function resolveEventText(request) {
+	let text = request.responseText;
+	if (request.status !== 200) {
+		const textParts = text.substr(text.indexOf('HTTP Status Code')).split('\n');
+		text = textParts[0] + '\r\n' + textParts[1];
+
+		if (request.status === 401) {
+			text = 'Please log in to Teamscale first: ' + teamscaleUrl + '\r\n\r\n' + text;
+		}
+	}
+
+	text = text.replace(/ : .:\/CQSE\/MSG_TIA:000/gm, ':');
```

Comment:
Maybe directly returning the `text.replace(/ : .:\/CQSE\/MSG_TIA:000/gm, ':')`?

Reply:
done